core/src/banking_stage/consumer.rs

```diff
@@ -392,14 +392,54 @@ impl Consumer {
         bank: &Arc<Bank>,
         txs: &[SanitizedTransaction],
         chunk_offset: usize,
+    ) -> ProcessTransactionBatchOutput {
+        // No filtering before QoS - transactions should have been sanitized immediately prior to this call
+        let pre_results = std::iter::repeat(Ok(()));
+        self.process_and_record_transactions_with_pre_results(bank, txs, chunk_offset, pre_results)
+    }
+
+    pub fn process_and_record_aged_transactions(
+        &self,
+        bank: &Arc<Bank>,
+        txs: &[SanitizedTransaction],
+        max_slot_ages: &mut [Slot],
+    ) -> ProcessTransactionBatchOutput {
+        // Need to filter out transactions since they were sanitized earlier.
+        // This means that the transaction may cross and epoch boundary (not allowed),
+        //  or account lookup tables may have been closed.
+        let pre_results = txs
+            .iter()
+            .zip(max_slot_ages)
+            .map(|(tx, max_slot_age)| {
+                if *max_slot_age < bank.slot() {
```

Comment:
hmm, come to think of it, all of txes could hit this condition across every epoch boundaries, assuming these transactions are sanitized against root bank? that would cause artificial tx processing stalls around them.

one potential solution is that sanitization should basically done with tip of given bank fork and check against bank.ancestors at the worker thread (i.e. here). but that means switching to different fork invalidates all the sanitized txes... (source of perf. hiccups)

seems harder than originally thought... 

alternatively, how about treating max_slot_ages as a hint from the scheduler thread? i.e. if this `if` condition is hit, resanitize the original `Packet` here on the fly and process normally (if it's sanitized correctly and is identical to the one sanitized at prior epoch's bank)

Reply:
> alternatively, how about treating max_slot_ages as a hint from the scheduler thread? i.e. if this if condition is hit, resanitize the original Packet here on the fly and process normally (if it's sanitized correctly and is identical to the one sanitized at prior epoch's bank)

Yeah, this is likely better than sending it back to scheduler for re-sanitization and using some sentinel value like I'm doing now.

Re-santiizing in worker @ epoch-crosses will lead to some overhead, but think it's somewhat unavoidable...i.e. we either always sanitize in worker, or we only do it when we need to. 

An optimist's take on it: we think of this not as "additional overhead at epoch-crosses", but "less overhead everywhere except epoch crosses".

Reply:
Would like some feedback on 3e1187079fe4b9365cee742a3315c7d0c8f0aaa7.

In the conversion from `ImmutableDeserializedPacket` -> `SanitizedTransaction`, the only meaningful sanitization (other than address lookup table checks) that uses features, is the pre-compile checks. So I do that on the worker side, if epoch is crossed, before checking ALTs.
I'm not super happy with it because this is kind of relying on that implementation.

I'm trying to avoid passing the Packets from Scheduler to ConsumeWorker though. And nothing will meaningfully change for these transactions - either the current `SanitizedTransaction` is valid and we don't need to re-allocate, or it is not, to the best of my knowledge.

Reply:
>  I'm not super happy with it because this is kind of relying on that implementation.

I second too. Any wild future feature activation could screw this abstraction leak of implementation details... How about serializing SanitizedTransaction and deserialize back into VersionedTransaction and sanitize it with bank at hand of worker?

for sigverify purpose, these byte conversion are guaranteed to be lossless and bijection.

so that you don't need pass Packets and i think these additional cost is acceptable because it's just for epoch boundaries.

Reply:
Added an internal resanitize fn: ca20a5e6708c75409d04664fab7f70f1bf696b3e

Made some comments there as well. I did not replace the original transaction, as sanitization should not affect the resulting transaction. We only need to check that sanitization checks still pass.

Reply:
(sorry for half-baked review; i gotta to bed...; i'll reply the rest open review threads tomorrow)

it took some time to figure out this twin of `SanitizedTransaction` code path drama as you know, lol

that said, i think https://github.com/solana-labs/solana/commit/ca20a5e6708c75409d04664fab7f70f1bf696b3e isn't actually calling the mentioned `verify_precompiles()`...

how about this simplification?: b9760bab6a0 in short, i think replay variant is future-proof and should be safe to use in the banking stage context _in theory and in practice_. :)

> ... (if it's sanitized correctly and is identical to the one sanitized at prior epoch's bank)

i still think comparing is important as commented in my commit